src/spatialdata/_core/operations/map.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from collections.abc import Iterable, Mapping
+from types import MappingProxyType
+from typing import Any, Callable
+
+import dask.array as da
+from dask.array.overlap import coerce_depth
+from datatree import DataTree
+from xarray import DataArray
+
+from spatialdata.models._utils import get_axes_names, get_channels, get_raster_model_from_data_dims
+from spatialdata.models.models import Labels2DModel, Labels3DModel, get_model
+from spatialdata.transformations import get_transformation
+
+__all__ = ["map_raster"]
+
+
+def map_raster(
+    data: DataArray | DataTree,
+    func: Callable[[da.Array], da.Array],
+    fn_kwargs: Mapping[str, Any] = MappingProxyType({}),
+    blockwise: bool = True,
+    depth: int | tuple[int, ...] | dict[int, int] | None = None,
+    chunks: tuple[tuple[int, ...], ...] | None = None,
+    c_coords: Iterable[int] | Iterable[str] | None = None,
+    dims: tuple[str, ...] | None = None,
+    transformations: dict[str, Any] | None = None,
+    **kwargs: Any,
+) -> DataArray:
+    """
+    Apply a function to raster data.
+
+    Parameters
+    ----------
+    data
+        The data to process. It can be a `DataArray` or `DataTree`. If it's a `DataTree`,
+        the function is applied to the first scale (full-resolution data).
+    func
+        The function to apply to the data.
+    fn_kwargs
+        Additional keyword arguments to pass to the function `func`.
+    blockwise
+        If `True`, distributed processing will be achieved with `dask.array.map_overlap`/`dask.array.map_blocks`,
+        otherwise the function is applied to the full data. If `False`, `depth` and `chunks` are ignored.
+    depth
+        If not `None`, distributed processing will be achieved with `dask.array.map_overlap`, otherwise with
+        `dask.array.map_blocks`. Specifies the overlap between chunks, i.e. the number of elements that each chunk
+        should share with its neighbor chunks. Please see `dask.array.map_overlap` for more information on the accepted
+        values.
+    chunks
+        Passed to `dask.array.map_overlap`/`dask.array.map_blocks` as `chunks`. Ignored if `blockwise` is `False`.
+        Chunk shape of resulting blocks if the function does not preserve the data shape. If not provided, the resulting
+        array is assumed to have the same chunk structure as the first input array.
+        E.g. ( (3,), (100,100), (100,100) ).
+    c_coords
+        The channel coordinates for the output data. If not provided, the channel coordinates of the input data are
+        used. It should be specified if the function changes the number of channels.
```

Comment:
Now it defaults to the channels in the input `data`, see:
        `c_coords = c_coords if c_coords is not None else get_channels(data)`

I think this makes sense, and feels a bit cleaner than setting it to `range(len(output_coords))`, because if you do not alter number of channels in `func`, you just want to copy channels from input to output I guess.

Reply:
I agree, I will change the code, and update docstring as follows:
```

    c_coords
        The channel coordinates for the output data. If not provided, the channel coordinates of the input data are
        used. If the callable `func` is expected to change the number of channel coordinates,
        this argument should be provided, otherwise will default to range(len(output_coords)).


```